demos/device_defender_for_aws/defender_demo.c

```diff
@@ -427,6 +447,36 @@ static BaseType_t collectDeviceMetrics( void )
         }
     }
 
+    /* Collect custom metrics. This demo sends this tasks stack high water mark
+     * as a number type custom metric and the current task ids as a list of
+     * numbers type custom metric. */
+    if( metricsCollectorStatus == MetricsCollectorSuccess )
+    {
+        vTaskGetInfo(
+            /* Query this task. */
+            NULL,
+            &pTaskStatus,
+            /* Include the stack high water mark value. */
+            pdTRUE,
+            /* Don't include the task state in the TaskStatus_t structure. */
+            0 );
+        tasksWritten = uxTaskGetSystemState( pTaskList, CUSTOM_METRICS_TASKS_ARRAY_SIZE, NULL );
+
+        if( tasksWritten == 0 )
+        {
+            metricsCollectorStatus = MetricsCollectorCollectionFailed;
+            LogError( ( "Failed to collect system state. uxTaskGetSystemState() failed due to insufficient buffer space.",
+                        metricsCollectorStatus ) );
+        }
+        else
+        {
+            for( i = 0; i < tasksWritten; i++ )
```

Comment:
Comment here about the operation of populating the task IDs in the custom metrics object from the queried system state will be helpful